anomalib/utils/sweep/helpers/inference.py

```diff
@@ -0,0 +1,147 @@
+"""Utils to help compute inference statistics."""
+
+# Copyright (C) 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+import time
+from pathlib import Path
+from typing import Dict, Iterable, List, Tuple, Union
+
+import numpy as np
+from omegaconf import DictConfig, ListConfig
+from torch.utils.data import DataLoader
+
+from anomalib import AnomalyModule
+from anomalib.deploy import OpenVINOInferencer, TorchInferencer
+
+
+class MockImageLoader:
+    """Create mock images for inference on CPU based on the specifics of the original torch test dataset.
+
+    Uses yield so as to avoid storing everything in the memory.
+
+    Args:
+        image_size (List[int]): Size of input image
+        total_count (int): Total images in the test dataset
+    """
+
+    def __init__(self, image_size: List[int], total_count: int):
+        self.total_count = total_count
+        self.image_size = image_size
+        self.image = np.ones((*self.image_size, 3)).astype(np.uint8)
+
+    def __len__(self):
+        """Get total count of images."""
+        return self.total_count
+
+    def __call__(self) -> Iterable[np.ndarray]:
+        """Yield batch of generated images.
+
+        Args:
+            idx (int): Unused
+        """
+        for _ in range(self.total_count):
+            yield self.image
+
+
+def get_meta_data(model: AnomalyModule, input_size: Tuple[int, int]) -> Dict:
+    """Get meta data for inference.
+
+    Args:
+        model (AnomalyModule): Trained model from which the metadata is extracted.
+        input_size (Tuple[int, int]): Input size used to resize the pixel level mean and std.
+
+    Returns:
+        (Dict): Metadata as dictionary.
+    """
+    meta_data = {
+        "image_threshold": model.image_threshold.value.cpu().numpy(),
+        "pixel_threshold": model.pixel_threshold.value.cpu().numpy(),
+        "stats": {},
```

Comment:
Do we need to add the `min` and `max` here as metadata, now that we also support min max normalization?